internal/feather.go

```diff
@@ -57,21 +58,161 @@ func convertReflectToArrowType(value reflect.Type) (arrow.DataType, error) {
+	case reflect.Int:
+		arrayValues := values.([]int)
+		convertedValues := []int64{}
+		for _, value := range arrayValues {
+			convertedValues = append(convertedValues, int64(value))
+		}
+		builder.(*array.Int64Builder).AppendValues(convertedValues, nullMask)
+	case reflect.Int8:
+		builder.(*array.Int8Builder).AppendValues(values.([]int8), nullMask)
+	case reflect.Int16:
+		builder.(*array.Int16Builder).AppendValues(values.([]int16), nullMask)
+	case reflect.Int32:
+		builder.(*array.Int32Builder).AppendValues(values.([]int32), nullMask)
+	case reflect.Int64:
+		builder.(*array.Int64Builder).AppendValues(values.([]int64), nullMask)
+	case reflect.Uint:
+		arrayValues := values.([]uint)
+		convertedValues := []uint64{}
+		for _, value := range arrayValues {
+			convertedValues = append(convertedValues, uint64(value))
+		}
+		builder.(*array.Uint64Builder).AppendValues(convertedValues, nullMask)
+	case reflect.Uint8:
+		builder.(*array.Uint8Builder).AppendValues(values.([]uint8), nullMask)
+	case reflect.Uint16:
+		builder.(*array.Uint16Builder).AppendValues(values.([]uint16), nullMask)
+	case reflect.Uint32:
+		builder.(*array.Uint32Builder).AppendValues(values.([]uint32), nullMask)
+	case reflect.Uint64:
+		builder.(*array.Uint64Builder).AppendValues(values.([]uint64), nullMask)
+	case reflect.Float32:
+		builder.(*array.Float32Builder).AppendValues(values.([]float32), nullMask)
+	case reflect.Float64:
+		builder.(*array.Float64Builder).AppendValues(values.([]float64), nullMask)
+	case reflect.String:
+		builder.(*array.LargeStringBuilder).AppendValues(values.([]string), nullMask)
+	case reflect.Bool:
+		builder.(*array.BooleanBuilder).AppendValues(values.([]bool), nullMask)
+	case reflect.Struct:
+		if elemType == reflect.TypeOf(time.Time{}) {
+			timeSlice := values.([]time.Time)
+			timestampSlice := make([]arrow.Timestamp, 0, len(timeSlice))
+			for _, t := range timeSlice {
+				timestampSlice = append(timestampSlice, arrow.Timestamp(t.UnixNano()))
+			}
+			builder.(*array.TimestampBuilder).AppendValues(timestampSlice, nullMask)
+		} else {
+			sBuilder, ok := builder.(*array.StructBuilder)
+			if !ok {
+				return errors.Errorf("internal error: expected struct builder, found %T", builder)
+			}
+
+			numFieldsReflect := slice.Index(0).NumField()
+			numFieldsArrow := sBuilder.NumField()
+			if numFieldsReflect != numFieldsArrow {
+				return errors.Errorf(
+					"expected number of fields in struct to match number of fields in Arrow struct schema, "+
+						"found %d fields in struct and %d fields in Arrow struct schema",
+					numFieldsReflect,
+					numFieldsArrow,
+				)
+			}
+
+			var columns []reflect.Value
+			for j := 0; j < numFieldsReflect; j++ {
+				firstStruct := slice.Index(0)
+				sliceType := reflect.SliceOf(firstStruct.Field(j).Type())
+				col := reflect.MakeSlice(sliceType, 0, slice.Len())
+				for i := 0; i < slice.Len(); i++ {
+					col = reflect.Append(col, slice.Index(i).Field(j))
```

Comment:
This will copy the values, so when this new slice is modified, the original slice will not be affected, so the values will get lost, I think? The test should catch this, but it looks like it's not currently asserting equality.